smartsim/_core/launcher/colocated.py

```diff
@@ -56,6 +56,8 @@ def write_colocated_launch_script(
         script_file.write("set -e\n\n")
 
         script_file.write("Cleanup () {\n")
+        script_file.write("echo error detected cleanup triggered\n")
```

Comment:
I think this message is not strictly true because the ``cleanup()`` method is triggered on any exit (including non-error exits.  We should at least clean this message up but also I'm curious if the group thinks we should always be printing this with the process ID. @ashao @al-rigazzi

Reply:
I think it would be good to add the process ID, to make post-mortem analysis of failed colocated orchestrators easier.